src/libcrun/linux.c

```diff
@@ -2483,6 +2478,11 @@ libcrun_set_mounts (struct container_entrypoint_s *entrypoint_args, libcrun_cont
   if (UNLIKELY (ret < 0))
     return ret;
 
+  // configure handler mounts
+  ret = libcrun_container_notify_handler (entrypoint_args, HANDLER_CONFIGURE_MOUNTS, container, rootfs, err);
```

Comment:
if we move this phase after we make the mount readonly, it is more difficult to setup mounts.

Can it be before the `if (def->root->readonly)` part?

Reply:
Sure let me try it out.

Reply:
Moving this before to `if (def->root->readonly)` causes krun to panic i think `krun`  needs `kvm` device to be mounted after `do_mount` not sure what is the exact reason for this yet. 
```console
thread '<unnamed>' panicked at 'Error creating the Kvm object: Error(2)', src/vmm/src/linux/vstate.rs:340:30
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fatal runtime error: failed to initiate panic, error 5
```
But keeping it here or after `do_mount` makes it work for both `mono` and `krun`.

I also have full stackstrace for this krun panic but that does look helpful so shared without `RUST_BACKTRACE=full`.

Reply:
I tried with `RUST_BACKTRACE=full` as well just sharing the stacktrace here.
```console
thread '<unnamed>' panicked at 'Error creating the Kvm object: Error(2)', src/vmm/src/linux/vstate.rs:340:30
stack backtrace:
   0:     0x7f216d9e3ebe - <unknown>
   1:     0x7f216d978bec - <unknown>
   2:     0x7f216d9e3641 - <unknown>
   3:     0x7f216d9e2fe5 - <unknown>
   4:     0x7f216d9e26a5 - <unknown>
   5:     0x7f216da01698 - <unknown>
   6:     0x7f216da0160c - <unknown>
   7:     0x7f216da015bd - <unknown>
   8:     0x7f216d977650 - <unknown>
   9:     0x7f216d979292 - <unknown>
  10:     0x7f216da2888a - <unknown>
  11:     0x7f216d9cec1d - <unknown>
  12:     0x5568e451cf89 - <unknown>
  13:     0x5568e44fdf50 - <unknown>
  14:     0x5568e4514c2a - <unknown>
  15:     0x5568e44fcb9c - <unknown>
  16:     0x5568e44ff665 - <unknown>
  17:     0x5568e44f7830 - <unknown>
  18:     0x5568e44f6e16 - <unknown>
  19:     0x7f216dd230b3 - <unknown>
  20:     0x5568e44f6f0e - <unknown>
  21:                0x0 - <unknown>
fatal runtime error: failed to initiate panic, error 5
```

Reply:
`Error(2)` is `ENOENT`, so it seems like `/dev/kvm` is not getting bind-mounted in the final context.

Reply:
After discussion with @giuseppe it was decided to move `HANDLER_CONFIGURE_AFTER_MOUNTS` just after we finalize container mounts so since i have amended the PR this should not be an issue anymore.